### PR TITLE
fix: upgrade dependencies

### PR DESCRIPTION
**Pull request by huimiu**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25386097
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25593617

### CODE REVIEW

File: templates/ts/message-extension-copilot/package.json.tpl [hunk 1]
```diff
@@ -25,15 +25,15 @@
         "adaptivecards-templating": "^2.3.1",
         "adaptivecards": "^3.0.1",
         "botbuilder": "^4.20.0",
-        "restify": "^10.0.0"
+        "restify": "^11.1.0"
     },
     "devDependencies": {
         "@types/restify": "^8.5.5",
-        "@types/node": "^16.0.0",
+        "@types/node": "^20.8.9",
```
Comment (hund030) [inline templates/ts/message-extension-copilot/package.json.tpl:32]: Does this require node@20?

Comment (huimiu) [reply templates/ts/message-extension-copilot/package.json.tpl:32]: While there isn't a strict requirement, considering this template doesn't have any other specific dependencies and @types/node is a dev dependency, I've tested it without issues after upgrading to node@20. To comprehensively address the issue mentioned in the bug, I'm planning to upgrade to node@20.

Comment (hund030) [reply templates/ts/message-extension-copilot/package.json.tpl:32]: Should we also update the engine in package.json?
https://github.com/OfficeDev/TeamsFx/blob/b1cd95889caaf7c41e86376a0d16ab209537ac21/templates/ts/message-extension-copilot/package.json.tpl#L6

Comment (huimiu) [reply templates/ts/message-extension-copilot/package.json.tpl:32]: Updated, thx! I'll double check with Xiaofu, as she's working on the node@20 upgrade for TTK.

**Review by hund030 — APPROVED** [review]

File: templates/ts/message-extension-copilot/package.json.tpl (hunk 1 above)
Comment (huimiu) [reply templates/ts/message-extension-copilot/package.json.tpl:32]: Confirmed with Xiaofu, the template needs to ensure that the engines has a minimum version of 16.

**Pull request merged**